### PR TITLE
Run through

### PR DESCRIPTION
**Pull request by nickramsbottom**

### CODE REVIEW

File: 06/README.md [hunk 1]
```diff
@@ -2,6 +2,8 @@ Now that you've written a function to pass a given test, and written a test for
 
 Create a file called `hello_test.go`, and write a test called TestHello, just like in previous exercises. Have it call the `Greeting()` function, and expect the result `"Howdy, Gopherinos!"`.
 
+// Why is the test function exported?
```
Comment (bitfield) [inline 06/README.md:5]: Because `go test` only runs functions of the form `TestXXX`: https://golang.org/pkg/testing/

File: 07/go.mod [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module hello
+
+go 1.13
```
Comment (bitfield) [inline 07/go.mod:3]: Go 1.14 is out!

File: 08/README.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ func TestAdd(t *testing.T) {
 	}{
 		{a: 1, b: 1, want: 2},
 		{a: 2, b: 2, want: 4},
-		{a: 6, b: 7, want: 12},
+		{a: 6, b: 7, want: 13},
```
Comment (bitfield) [inline 08/README.md:23]: Well done for spotting the (not) deliberate mistake...

File: 10/calculator.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+// Multiply two numbers
+func Multiply(a, b int) int {
+	return a * b
+}
+
+// Divide two numbers
+func Divide(a, b int) (int, error) {
+	if b == 0 {
+		return 0, errors.New("cannot divide by zero")
```
Comment (bitfield) [inline 10/calculator.go]: It's always helpful to return the invalid data:

```suggestion
		return 0, fmt.Errorf("division by zero: %d, %d", a, b)
```

File: 10/calculator_test.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+		{a: -3, b: 1, want: -3, errExpected: false},
+		{a: -12, b: 6, want: -2, errExpected: false},
+		{a: 28, b: -7, want: -4, errExpected: false},
+		{a: 1, b: 0, want: 0, errExpected: true},
+	}
+	for _, testCase := range testCases {
+		got, err := calculator.Divide(testCase.a, testCase.b)
+
+		if err == nil && testCase.errExpected {
+			t.Errorf("Expected error for Divide(%d, %d)", testCase.a, testCase.b)
```
Comment (bitfield) [inline 10/calculator_test.go]: This should be `t.Fatalf`; `t.Errorf` reports the error, but allows the test to continue. Often we just want to bail out of the test case completely, so we'd use `t.Fatalf` for this.

**Review by bitfield — APPROVED** [review]
Great job! Clearly this was much too easy for you and you need the forthcoming (harder) series of exercises on data structures 😁

File: 07/go.mod (hunk 2 above)
Comment (nickramsbottom) [reply 07/go.mod:3]: Nice, I'll update my local!

File: 06/README.md (hunk 1 above)
Comment (nickramsbottom) [reply 06/README.md:5]: ahhh, is this why the package names are also different so you don't expose the test functions as well?

**Comment by nickramsbottom** [comment]
Thank you for your review! I'd be more than enthusiastic to do the data structures or any other exercises. Data structures in particular is an area I want to delve into more.

I really liked the introductions to the first few exercises. I find learning objectives helpful for keywords to search for in the documentation/book and making sure I've understood all of the intended topics.

I assumed the divide function should only result in whole numbers. Could this case be added to the error handling exercise?

File: 06/README.md (hunk 1 above)
Comment (bitfield) [reply 06/README.md:5]: Yes, it's a good idea to always test from _outside_ the package. One reason is that you get to be the first user of your package's API! If it doesn't make sense in the test, it probably won't make sense to other users.

Sometimes there really are things that you can only test from inside, but if we've designed the package well, there shouldn't be very many of these.

**Comment by bitfield** [comment]
Yes, that's a good point. I need to update the exercise to address that, thank you.

File: 06/README.md (hunk 1 above)
Comment (nickramsbottom) [reply 06/README.md:5]: Nice! I'll make sure to do that when I attempt a project

**Pull request merged**